### PR TITLE
Create pass through nse

### PR DESCRIPTION
**Pull request by panyuenlau**
This PR is for [NSMNSE-3](https://jira-eng-sjc4.cisco.com/jira/browse/NSMNSE-3).

Proof of connectivity between NSC and NSE through ICMP-responder NetworkService works:

ucnf-nse network interface addresses:
Screen Shot 2020-12-07 at 3 14 16 PM

NSC network interface info and pings from NSC to ucnf-nse:
Screen Shot 2020-12-07 at 3 14 26 PM

Proof of `universal-cnf-vppagent` image is created and ready to be pushed to docker hub in the circleCI pipeline:
Screen Shot 2020-12-07 at 10 54 49 AM

### CODE REVIEW

File: deployments/helm/ucnf-nse/values.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+nsm:
+  serviceName: ucnf-nse-service
+
+global:
+
+replicaCount: 1
+
+image:
+  repository: tiswanso
```
Comment (snowbldr) [inline deployments/helm/ucnf-nse/values.yaml]: this should probably be cisco-app-networking, @tiswanso thoughts?

File: pkg/universal-cnf/ucnf/ucnf_nse.go [hunk 2]
```diff
@@ -36,7 +36,9 @@ func NewUcnfNse(configPath string, verify bool, backend config.UniversalCNFBacke
 		}
 	}()
 
-	err = nseconfig.NewConfig(yaml.NewDecoder(f), cnfConfig)
+	decodedYAML := yaml.NewDecoder(f)
+	decodedYAML.SetStrict(true) // enable strict decoding to help debug potential issues for decoding YAMLs
+	err = nseconfig.NewConfig(decodedYAML, cnfConfig)
 	if err != nil {
 		logrus.Warningf("NSE config errors: %v", err)
```
Comment (ondrej-fabry) [inline pkg/universal-cnf/ucnf/ucnf_nse.go:50]: Should we keep just warning log here when decoding fails and not rather return error? @tiswanso is there a reason why we would want to continue execution when the config is not valid?

**Review by snowbldr — APPROVED** [review]
Tentatively approved pending Tim's verification of the image repository in the values.yaml

File: pkg/universal-cnf/ucnf/ucnf_nse.go (hunk 2 above)
Comment (tiswanso) [reply pkg/universal-cnf/ucnf/ucnf_nse.go:50]: it'd be best to make the `SetStrict` is controllable via an env var

**Review by CosminNechifor — APPROVED** [review]

**Review by snowbldr — APPROVED** [review]

File: build/nse/universal-cnf/Makefile [hunk 3]
```diff
@@ -5,5 +5,6 @@ AUX_CONTAINERS =
 PODS =
 NETWORK_SERVICES =
 CHECK =
+VPP_AGENT=ligato/vpp-agent:v3.2.0
```
Comment (tiswanso) [inline build/nse/universal-cnf/Makefile:8]: The root nsm-nse/Makefile sets this already.  This shouldn't need to be set here.

**Comment by tiswanso** [comment]
Please submit this against wwwin-github.cisco.com/cisco-app-networking/nsm-nse and add modify the Jenkinsfile content for the container build part.

**Comment by tiswanso** [comment]
@panyuenlau -- per the webex teams conversation that this is superseded by #70, I'm closing this.

File: build/nse/universal-cnf/Makefile (hunk 3 above)
Comment (ondrej-fabry) [reply build/nse/universal-cnf/Makefile:8]: @tiswanso Actually there is a Makefile bug that makes the value of VPP_AGENT be overwritten by the kiknos NSE where it is set to kiknos agent.. and this value is then passed most likely via the for each loops and includes in makefiles.. and causing the the universal-cnf NSE image to use wrong base image (from kiknos).. which is not obvious from user perspective at all.. It tooks us a while to debug this behaviour with Darren

**Pull request closed without merging**